dashboard/lib/services/user/password_resetter.rb

```diff
@@ -1,10 +1,11 @@
 module Services
   module User
     class PasswordResetter < Services::Base
-      attr_reader :email
+      attr_reader :email, :username
 
-      def initialize(email:)
+      def initialize(email: nil, username: nil)
```

Comment:
I'd highly recommend creating two separate service objects for password resets — one by `email` and one by `username`. The new `username` argument introduces enough complexity and distinct differences to justify this refactoring, in my opinion.

Reply:
I would highly recommend creating two separate service objects for password resets — one by email and one by username. The new username argument introduces enough complexity and distinct differences to justify this refactoring, in my opinion.